static/js/settings_notifications.js

```diff
@@ -76,10 +87,37 @@ exports.desktop_icon_count_display_values = {
     },
 };
 
+function rerender_ui() {
+    const unmatched_streams_table = $("#stream-specific-notify-table").expectOne();
+    const unmatched_streams = stream_data.get_unmatched_streams_for_notification_settings();
+    const parent_container = $('#stream-specific-notify-settings').expectOne();
+
+    list_render.create(unmatched_streams_table, unmatched_streams, {
+        name: "unmatched-streams-list",
+        modifier: function (unmatched_streams) {
+            return render_stream_specific_notif_row({
+                unmatched_streams: unmatched_streams,
+                stream_specific_notification_settings: stream_specific_notification_settings,
+                push_notification_tooltip: exports.all_notifications.push_notification_tooltip,
+                page_params: page_params,
+            });
+        },
+        parent_container: parent_container,
+    }).init();
+
+    if (unmatched_streams.length === 0) {
+        parent_container.css("display", "none");
+    } else {
+        parent_container.css("display", "block");
+    }
+}
+
 function change_notification_setting(setting, setting_data, status_element) {
     const data = {};
     data[setting] = JSON.stringify(setting_data);
     settings_ui.do_settings_change(channel.patch, '/json/settings/notifications', data, status_element);
+    notifications.handle_global_notification_updates(setting, setting_data);
```

Comment:
What's the story behind these changes? Only for this line. Is this related to this commit or is it independent?

Reply:
This line is required for real-time synchronization for the case when we update the global notification settings, so that the unmatched streams table gets updated (more specifically, `stream_data.get_unmatched_streams_for_notification_settings()` does not return the updated data)

I'm not too sure whether there were any bugs due to not including this line, but this was noticed due to this new UI, so i guess this should be a part of this commit?

Reply:
I think this call belongs in the `server_events_dispatch` code path (like any other update of UI based on changes submitted to the server).

You'll see that this is broken if you run two browser windows and make changes in one but not the other.

Reply:
Oh I see, my mistake.
This line was already present.
(Shifted the UI updates to the proper function which is called from the `server_events_dispatch` code path.)